src/Neo/SmartContract/ApplicationEngine.cs

```diff
@@ -399,13 +407,42 @@ internal override void UnloadContext(ExecutionContext context)
         /// <returns>The engine instance created.</returns>
         public static ApplicationEngine Create(TriggerType trigger, IVerifiable container, DataCache snapshot, Block persistingBlock = null, ProtocolSettings settings = null, long gas = TestModeGas, IDiagnostic diagnostic = null)
         {
+            var index = persistingBlock?.Index ?? NativeContract.Ledger.CurrentIndex(snapshot);
```

Comment:
@shargon 
Object reference not set to an instance of an object. 

on test `Neo.UnitTests.SmartContract.UT_NotifyEventArgs.TestIssue3300`

Reply:
Snapshot is null?

Reply:
yes

Reply:
Would this work for current `CurrentIndex`?
```csharp
snapshot?[CreateStorageKey(Prefix_CurrentBlock)].GetInteroperable<HashIndexState>()?.Index ?? 0; 
```

Reply:
We should not change the logic in native contracts for this